gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnAppLauncher.java

```diff
@@ -155,6 +155,8 @@
 
   private static final String GOBBLIN_YARN_APPLICATION_TYPE = "GOBBLIN_YARN";
 
+  private static final String FAIL_RECONNECTABLE_APP = "failReconnectableAppMaster";
```

Comment:
how about naming this to "gobblin.yarn.app.try.reconnect.before.launch" or "gobblin.yarn.app.master.reconnect" and placing it at `GobblinYarnConfigurationKeys` with default true ?